app/src/androidTest/java/com/android/unio/ui/saved/SavedTest.kt

```diff
@@ -0,0 +1,130 @@
+import com.android.unio.ui.navigation.NavigationAction
+import com.android.unio.ui.navigation.TopLevelDestination
+import com.google.firebase.Timestamp
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.just
+import io.mockk.runs
+import io.mockk.spyk
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@HiltAndroidTest
+class SavedTest {
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @get:Rule val hiltRule = HiltAndroidRule(this)
+
+  private lateinit var userViewModel: UserViewModel
+
+  // Mock event repository to provide test data.
+  @MockK private lateinit var eventRepository: EventRepositoryFirestore
+
+  @MockK private lateinit var userRepository: UserRepositoryFirestore
+
+  @MockK private lateinit var navigationAction: NavigationAction
+
+  @MockK private lateinit var imageRepository: ImageRepositoryFirebaseStorage
+
+  private lateinit var eventViewModel: EventViewModel
+  private lateinit var searchViewModel: SearchViewModel
+
+  @MockK(relaxed = true) private lateinit var searchRepository: SearchRepository
+
+  private lateinit var eventList: List<Event>
+  private lateinit var eventListFollowed: List<Event>
+
+  @Before
+  fun setUp() {
+    MockKAnnotations.init(this)
+    hiltRule.inject()
+    searchViewModel = spyk(SearchViewModel(searchRepository))
+    every { navigationAction.navigateTo(any(TopLevelDestination::class)) } returns Unit
+    every { navigationAction.navigateTo(any(String::class)) } returns Unit
+
+    every { userRepository.init(any()) } just runs
+    every { eventRepository.init(any()) } just runs
+    userViewModel = spyk(UserViewModel(userRepository))
+    val user = MockUser.createMockUser()
+    val asso = MockAssociation.createMockAssociation()
+
+    every { userRepository.updateUser(user, any(), any()) } answers
+        {
+          val onSuccess = args[1] as () -> Unit
+          onSuccess()
+        }
+    userViewModel.addUser(user, {})
+    every { userRepository.init(any()) } just runs
+    every { eventRepository.getEvents(any(), any()) } answers
+        {
+          val onSuccess = args[0] as (List<Event>) -> Unit
+          onSuccess(eventList)
+        }
+    eventViewModel = EventViewModel(eventRepository, imageRepository)
+
+    val field =
+        userViewModel.javaClass.getDeclaredMethod("setFollowedAssociations", List::class.java)
+    field.isAccessible = true
+    field.invoke(userViewModel, listOf(asso.uid))
+
+    eventList =
+        listOf(
+            MockEvent.createMockEvent(organisers = listOf(asso)),
+            MockEvent.createMockEvent(title = "I am different", startDate = Timestamp.now()))
+
+    val eventField = eventViewModel.javaClass.getDeclaredMethod("setEvents", List::class.java)
+    eventField.isAccessible = true
+    eventField.invoke(eventViewModel, eventList)
```

Comment:
Here I get it more easily, as we made it harder to add events to eventViewModel, but I think we will have to refactor EventVIewModel a bit in a next PR

Reply:
Indeed, there is no reason to manually inject the event list when you have already mocked the repository's `getEvents` method. Additionally, tests shouldn't depend on the internal behavior of a view model, as it can be subject to change at any time. Consider removing this.

Reply:
I do not agree on this. If you want to test the behavior of the view model while using the UI, this is needed